### ISSUE TITLE
"message":"New version app has been published"

### ISSUE THREAD

**Issue by majurgens** [lostfields/python-panasonic-comfort-cloud#103]
Something has changed again

```
--- Error finding meta_tag
Traceback (most recent call last):
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud.py", line 5, in <module>
    __main__.main()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/__main__.py", line 200, in main
    session.login()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 36, in login
    state = super().login()
            ^^^^^^^^^^^^^^^
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 281, in login
    self._get_new_token()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 251, in _get_new_token
    check_response(response, 'get_acc_client_id', 200)
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 25, in check_response
    raise exceptions.ResponseError(
pcomfortcloud.exceptions.ResponseError: (get_acc_client_id: Expected status code 200, received: 401: {
  "code":4106,
  "message":"New version app has been published"
}
```

Changing X_APP_VERSION to 
`X_APP_VERSION = "1.22.0"`
just gives a different error of:

```
--- Error finding meta_tag
Traceback (most recent call last):
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud.py", line 5, in <module>
    __main__.main()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/__main__.py", line 200, in main
    session.login()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 36, in login
    state = super().login()
            ^^^^^^^^^^^^^^^
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 281, in login
    self._get_new_token()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 251, in _get_new_token
    check_response(response, 'get_acc_client_id', 200)
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 25, in check_response
    raise exceptions.ResponseError(
pcomfortcloud.exceptions.ResponseError: (get_acc_client_id: Expected status code 200, received: 412: {
  "code":41201,
  "message":"Precondition Failed"
}
```

**Comment by xkram15** [comment]
Same here, starting today I am getting only mentioned error. :-(

**Comment by chamsters** [comment]
Same issue here. Trying anything over 1.22.0 also gives the "New version app has been published" error.

**Comment by chamsters** [comment]
Seems related - https://github.com/sockless-coding/panasonic_cc/issues/331

**Comment by majurgens** [comment]
Oh, they appear to have added another annoying feature. They now email me every time I login ie run the python script, or attempt to login with a message along the lines of "You have logged in to your Panasonic ID."

**Comment by lostfields** [comment]
A login procedure shouldn't be necessary when using offline access with OAuth2, so, every time we actually use the username/password they will send an email? If we can actually always use the refresh token correctly we shouldn't receive an email, right?

If both cases sends an email this is an active way to get rid of third-parties. 

Anyway, since they are blocking requets now, they have changed something again to detect third-party requests.

**Comment by majurgens** [comment]
I also get emails when I login via their website

**Comment by xkram15** [comment]
Hello,
out of curiosity - where and how do you "login via their website"?
I have three units "Panasonic Etherea" AC and so far I was aware only remote control by their  "Panasonic Comfort Cloud App" from a mobile. I tried to Google but still cannot find what are you referring. From where I can control my units from a website?

Thanks

**Comment by majurgens** [comment]
The "login via website" is only to manage the identity. Link is a few posts earlier

**Comment by chamsters** [comment]
Looks like there's a solution over at https://github.com/sockless-coding/panasonic_cc/releases/tag/2024.11.0, how can we incorporate into this project?

**Comment by Olifant1990** [comment]
Solved with https://github.com/sockless-coding/panasonic_cc/pull/336.
I can confirm it is working again!

**Comment by mchacher** [comment]
good news. @Olifant1990 can you propose a pull request with the fix?

**Comment by Olifant1990** [comment]
@mchacher no sorry, not using this specific github, but confirmed the linked fix is working. So it shouldn't be difficult for you too ;)

**Comment by majurgens** [comment]
If I use the code from this new PR I am getting the following error:
```
Traceback (most recent call last):
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud.py", line 5, in <module>
    __main__.main()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/__main__.py", line 200, in main
    session.login()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/session.py", line 36, in login
    state = super().login()
            ^^^^^^^^^^^^^^^
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 307, in login
    self._get_new_token()
  File "/opt/hvac/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 271, in _get_new_token
    "x-cfc-api-key": self._get_api_key(timestamp, self._token["access_token"]),
                                                  ~~~~~~~~~~~^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

```

**Comment by xkram15** [comment]
Exactly the same - with a change from @mortenf I am getting :
...
...
  File "/home/pi/pc/python-panasonic-comfort-cloud/pcomfortcloud/authentication.py", line 271, in _get_new_token
    "x-cfc-api-key": self._get_api_key(timestamp, self._token["access_token"]),
TypeError: 'NoneType' object is not subscriptable

**Comment by mortenf** [comment]
I'm sorry, that should now be fixed in the PR. The error shows up when no token is saved/reused, which is why I didn't notice it.

**Comment by lostfields** [comment]
Fixed by #104
